samples/client/petstore/lua/api/store_api.lua

```diff
@@ -0,0 +1,229 @@
+local http_request = require "http.request"
+local http_util = require "http.util"
+local dkjson = require "dkjson"
+local basexx = require "basexx"
+
+-- model import
+local petstore_store_api = require "petstore.api.store_api"
+
+local petstore= {}
+local petstore_mt = {
+	__name = "store_api";
+	__index = petstore;
+}
+
+local function new_store_api(host, basePath, schemes)
+	local schemes_map = {}
+	for _,v in ipairs(schemes) do
+		schemes_map[v] = v
+	end
+	local default_scheme = schemes_map.https or schemes_map.http
+	return setmetatable({
+		host = host;
+		basePath = basePath or "http://petstore.swagger.io/v2";
+		schemes = schemes_map;
+		default_scheme = default_scheme;
+		http_username = nil;
+		http_password = nil;
+		api_key = {};
+		access_token = nil;
+	}, petstore_mt)
+end
+
+function store_api:delete_order(order_id)
+	local req = http_request.new_from_uri({
+		scheme = self.default_scheme;
+		host = self.host;
+		path = string.format("%s/store/order/%s",
+			self.basePath, order_id);
+	})
+
+	-- set HTTP verb
+	req.headers:upsert(":method", "DELETE")
+
+	-- TODO: create a function to select proper content-type
+	-- ref: https://github.com/swagger-api/swagger-codegen/pull/6252#issuecomment-321199879
+	--local var_accept = { "application/xml", "application/json" }
+	req.headers:upsert("content-type", "application/xml")
+
+
+
+	-- make the HTTP call
+	local headers, stream, errno = req:go()
+	if not headers then
+		return nil, stream, errno
+	end
+	local http_status = headers:get(":status")
+	if http_status:sub(1,1) == "2" then
+		return nil, headers
+	else
+		local body, err, errno2 = stream:get_body_as_string()
+		if not body then
+			return nil, err, errno2
+		end
+		stream:shutdown()
+		-- return the error message (http body)
+		return nil, http_status, body
+	end
+end
+
+function store_api:get_inventory()
+	local req = http_request.new_from_uri({
+		scheme = self.default_scheme;
+		host = self.host;
+		path = string.format("%s/store/inventory",
+			self.basePath);
+	})
+
+	-- set HTTP verb
+	req.headers:upsert(":method", "GET")
+
+	-- TODO: create a function to select proper content-type
+	-- ref: https://github.com/swagger-api/swagger-codegen/pull/6252#issuecomment-321199879
+	--local var_accept = { "application/json" }
+	req.headers:upsert("content-type", "application/json")
+
+
+	-- api key in headers 'api_key'
+	req.headers:upsert("api_key", api_key['api_key'])
+
+	-- make the HTTP call
+	local headers, stream, errno = req:go()
+	if not headers then
+		return nil, stream, errno
+	end
+	local http_status = headers:get(":status")
+	if http_status:sub(1,1) == "2" then
+		local body, err, errno2 = stream:get_body_as_string()
+		-- exception when getting the HTTP body
+		if not body then
+			return nil, err, errno2
+		end
+		stream:shutdown()
+		local result, _, err3 = dkjson.decode(body)
+		-- exception when decoding the HTTP body
+		if result == nil then
+			return nil, err3
+		end
+		return petstore_{}.cast(result), headers
```

Comment:
Right. That method returns an array of object. Not sure how we can deserialize the result back into an array of object in Lua